mysql-test/suite/mysql_migration/mysql_migration.test

```diff
@@ -0,0 +1,206 @@
+#    use a workaround to also capture the \n as a capturing group and put back into the
+#    replacement string
+# 2. When masking the container id, seems tat --regex_regex does not allow pattern matching
+#    preceding lines. Thus do a bind search on anything continues string longer than 40.
+--replace_regex /(?<=sha256:).*(\n)/<MASKED_CID>\1/  /^[a-zA-Z0-9]{40,}/<MASKED_CID>/
+--exec $POPULATE_MYSQL_DATADIR $TEST_MYSQL_VER $MYSQLTEST_VARDIR/tmp $MYSQL_COMMANDS 2>&1
+
+# Restart the server from the populated pseudo MySQL data directory
+let $restart_parameters=--datadir=$MYSQLTEST_VARDIR/tmp/mysql;
+--source include/restart_mysqld.inc 
+
+--echo #
+--echo # Run mariadb-upgrade
+--echo #
+
+# In fact some incompatibility errors may already appear during the upgrade run
+--exec $MYSQL_UPGRADE --force 2>&1
+
+# Restart the server and run mariadb-upgrade one more time to make sure that errors appeared
+# in the last run are resolved (e.g. column number mismatch for mysql sys.xxx views)
+--source include/restart_mysqld.inc 
+--exec $MYSQL_UPGRADE --force 2>&1
+
+--echo #
+--echo # Test that MariaDB can access tables partitioned with "native" InnoDb partition handler
+--echo #
+
+#
+# MariDB doc:
+#   MariaDB 10.3 does not support MySQL 5.7's “native” InnoDB partitioning handler.
+#
+# Please make sure the coresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_partition_db1;
+
+SHOW TABLES;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_partition_t1;
+
+# Confirm that MariaDB can still apply partitioning to the table as usual
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+ALTER TABLE idb_partition_t1
+    PARTITION BY HASH(id)
+    PARTITIONS 8;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_partition_t1;
+
+--echo #
+--echo # Test that MariaDB can access tables compressed using MySQL punch-hole compression
+--echo #
+
+#
+# Please make sure the corresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_page_comp_db1;
+
+SHOW TABLES;
+
+--error ER_TABLE_CORRUPT # Allow the test to continue. Remove once this incompatibility is resolved
+select count(*) as EXPECT_3000 from idb_page_comp_t1;
+
+
+--echo #
+--echo # Test that MariaDB can access tables encrypted in MySQL with keyring_file plug-in
+--echo #
+
+#
+# MariDB doc:
+#   MariaDB 10.3 implements InnoDB encryption in a different way to MySQL 5.7
+#
+# Please make sure the corresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_encryption_db1;
+
+SHOW TABLES;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_encryption_t1;
```

Comment:
Do we really want to implement the MySQL 5.7 way of encrypting tables? Such tables could be rebuilt before upgrading.

If MySQL 5.7 can combine encryption and compression, that should be tested as well. Likewise, encryption with `ROW_FORMAT=COMPRESSED` and `KEY_BLOCK_SIZE` of `1`, `2`, `4` needs to be tested.

Reply:
Right now, many MySQL users are going to look at [the docs on upgrading from MySQL → MariaDB](https://mariadb.com/kb/en/upgrading-from-mysql-to-mariadb/) and see…

  * "you can **_in most cases_** just uninstall MySQL [5.7] and install MariaDB and you are good to go. There is no need to dump and restore databases. As with any upgrade, we recommend making a backup of your data beforehand."

If MySQL users have encrypted tables, this will be a bad surprise, because those tables _won't work_ after migrating to MariaDB.

> Such tables could be rebuilt before upgrading. Do we really want to implement the MySQL 5.7 way of encrypting tables? 

That I don't know, but…

Surely the answer **is not** to continue surprising MySQL users with "oops, your table is corrupt" after they attempt to migrate to MariaDB, right?

That kind of uncertainty is precisely what will discourage a lot of users from switching.

Another possibilities would be to give a substantially more informative error than `ER_NO_SUCH_TABLE_IN_ENGINE`. How about at least try to enumerate and explain the on-disk files? Perhaps something like `ER_INCOMPATIBLE_TABLE_FORMAT`:

  * "This table was produced by a version of MySQL which stores it in an on-disk format that this version of MariaDB does not directly support, because $REASON. See https://mariadb.com/kb/en/upgrading-from-mysql-to-mariadb#known-incompatibilities. Please revert to the old engine version and try converting the table to a supported format before migrating to MariaDB."